src/main/java/org/jabref/JabRefExecutorService.java

```diff
@@ -87,6 +87,20 @@ public boolean executeAndWait(Callable command) {
         }
     }
 
+    /**
+     * This method allows to execute a callable task that provides a return value after the calculation is done.
+     * @param command The task to execute.
+     * @return A Future object that provides the returning value.
+     */
+    public <T> Future<T> executeAndReturn(Callable<T> command) {
+        if (command == null) {
```

Comment:
Why this? Can't we just put `Objects.requireNonNull(command)`? I think, Item 49 of [Effective Java, 3rd edition](https://www.amazon.de/Effective-Java-Third-Joshua-Bloch/dp/0134685997?tag=jabref-21).

Reply:
I did the null check this way because every other method in the class checked for null exactly the same way. In this case we should consider changing all other methods to `Objects.requireNonNull(...)`

Reply:
Yeah, this is the preferred way. Hope, nothing breaks if this is done at other places. 😇